src/Roslyn.Diagnostics.Analyzers/Core/SymbolIsBannedAnalyzer.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Threading;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Roslyn.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed partial class SymbolIsBannedAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string BannedSymbolsFileName = "BannedSymbols.txt";
+
+        internal static readonly DiagnosticDescriptor SymbolIsBannedRule = new DiagnosticDescriptor(
+            id: RoslynDiagnosticIds.SymbolIsBannedRuleId,
+            title: RoslynDiagnosticsAnalyzersResources.SymbolIsBannedTitle,
+            messageFormat: RoslynDiagnosticsAnalyzersResources.SymbolIsBannedMessage,
+            category: "ApiDesign",
+            defaultSeverity: DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: RoslynDiagnosticsAnalyzersResources.SymbolIsBannedDescription,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        internal static readonly DiagnosticDescriptor DuplicateBannedSymbolRule = new DiagnosticDescriptor(
+            id: RoslynDiagnosticIds.DuplicateBannedSymbolRuleId,
+            title: RoslynDiagnosticsAnalyzersResources.DuplicateBannedSymbolTitle,
+            messageFormat: RoslynDiagnosticsAnalyzersResources.DuplicateBannedSymbolMessage,
+            category: "ApiDesign",
+            defaultSeverity: DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: RoslynDiagnosticsAnalyzersResources.DuplicateBannedSymbolDescription,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(SymbolIsBannedRule, DuplicateBannedSymbolRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            // TODO: Make the analyzer thread-safe.
+            //context.EnableConcurrentExecution();
```

Comment:
Symbols your implementation is thread safe, any specific reason to not enable concurrent callbacks?

Reply:
Copy/pasta from the public API analyzer.